### ISSUE TITLE
Can't extract Marshmallow ROMs

### ISSUE THREAD

**Issue by h2o64** [xpirt/sdat2img#6]
Hi !
here my "extract log"
Skipping zero command.
Copying 32767 blocks into position 0...
Copying 2 blocks into position 32768...
Copying 2 blocks into position 32825...
Copying 32226 blocks into position 33307...
Copying 2 blocks into position 65536...
Copying 2 blocks into position 98304...
Copying 2 blocks into position 98361...
Copying 6452 blocks into position 98843...
Copying 2 blocks into position 131072...
Copying 2 blocks into position 163840...
Copying 2 blocks into position 163897...
Copying 2 blocks into position 196608...
Copying 17949 blocks into position 197090..

When I try to mount system.img :
mount: wrong fs type, bad option, bad superblock on /dev/loop0,
missing codepage or helper program, or other error

   In some cases useful info is found in syslog - try
   dmesg | tail or so.
On my dmseg
[13992.612662] EXT4-fs (loop0): bad geometry: block count 215040 exceeds size of device (215039 blocks)
[13994.338220] EXT4-fs (loop0): bad geometry: block count 215040 exceeds size of device (215039 blocks)
[13999.630333] EXT4-fs (loop0): bad geometry: block count 215040 exceeds size of device (215039 blocks)
[14080.991588] EXT4-fs (loop0): bad geometry: block count 215040 exceeds size of device (215039 blocks)
[14084.488430] EXT4-fs (loop0): couldn't mount as ext2 due to feature incompatibilities
[14119.141050] EXT4-fs (loop0): mounted filesystem with ordered data mode. Opts: (null)

It's on a 6.0 firmware.. Any clue ?

To fix this I have to use resiz2fs and resize it to a one block less image

**Comment by proDOOMman** [comment]
The same here:

```
$ ~/bin/sdat2img.py ./system.transfer.list ./system.new.dat ./system.img
Skipping zero command.
Copying 32767 blocks into position 0...
Copying 2 blocks into position 32768...
Copying 2 blocks into position 32833...
Copying 32212 blocks into position 33323...
Copying 2 blocks into position 65536...
Copying 2 blocks into position 98304...
Copying 2 blocks into position 98369...
Copying 32208 blocks into position 98859...
Copying 2 blocks into position 131072...
Copying 2 blocks into position 163840...
Copying 2 blocks into position 163905...
Copying 3982 blocks into position 164395...
Copying 2 blocks into position 196608...
Copying 2 blocks into position 229376...
Copying 2 blocks into position 229441...
Copying 19924 blocks into position 229931...

Done! Output image: /tmp/cm-13-20151028-UNOFFICIAL-falcon/system.img

$ sudo mount -o loop system.img system_extracted 
mount: wrong fs type, bad option, bad superblock on /dev/loop0,
       missing codepage or helper program, or other error
       In some cases useful info is found in syslog - try
       dmesg | tail or so.

$ dmesg | tail -n 1
[ 3429.588339] EXT4-fs (loop0): bad geometry: block count 249856 exceeds size of device (249855 blocks)
```

I'm trying to unpack this rom: http://forum.xda-developers.com/moto-g/development/rom-unofiicial-cyanogenmod-13-moto-g-t3229272

**Comment by zaphod-b** [comment]
still doesnt work. I tried it in windows and ubuntu using a cm13 built rom and output was exactly as above.

**Comment by xpirt** [comment]
Should be fixed with latest commit: ce30ec91fb74f0d9aed2ece334d663c40ddb4d3e by @pawitp.
